src/frontend/src/lib/components/swap/SwapProvider.svelte

```diff
@@ -1,81 +1,97 @@
 	import ModalValue from '$lib/components/ui/ModalValue.svelte';
 	import { i18n } from '$lib/stores/i18n.store';
 	import {
 		SWAP_AMOUNTS_CONTEXT_KEY,
 		type SwapAmountsContext
 	} from '$lib/stores/swap-amounts.store';
-	import type { OisyDappDescription } from '$lib/types/dapp-description';
 	import type { OptionString } from '$lib/types/string';
-	import type { ProviderFee } from '$lib/types/swap';
+	import { SwapProvider } from '$lib/types/swap';
 	import { replacePlaceholders } from '$lib/utils/i18n.utils';
 	import { UrlSchema } from '$lib/validation/url.validation';
 
 	const { store: swapAmountsStore } = getContext<SwapAmountsContext>(SWAP_AMOUNTS_CONTEXT_KEY);
 
-	let route = $derived<string[] | undefined>($swapAmountsStore?.swapAmounts?.route);
-	let networkFee = $derived<ProviderFee | undefined>($swapAmountsStore?.swapAmounts?.networkFee);
-	let liquidityFees = $derived<ProviderFee[] | undefined>(
-		$swapAmountsStore?.swapAmounts?.liquidityFees
-	);
 	let displayURL = $state<OptionString>(null);
-	let kongSwapDApp = $derived<OisyDappDescription | undefined>(
-		dAppDescriptions.find(({ id }) => id === 'kongswap')
+
+	let selectedProvider = $derived($swapAmountsStore?.selectedProvider);
+	let isBestRate = $derived(selectedProvider?.provider === $swapAmountsStore?.swaps[0]?.provider);
+	let swapDApp = $derived(
+		dAppDescriptions.find(
+			({ id }) => id === $swapAmountsStore?.selectedProvider?.provider.toLowerCase()
+		)
 	);
 
 	$effect(() => {
-		if (nonNullish(kongSwapDApp)) {
-			const parsed = UrlSchema.safeParse(kongSwapDApp.website);
+		if (nonNullish(swapDApp)) {
+			const parsed = UrlSchema.safeParse(swapDApp.website);
 			if (parsed.success) {
 				const url = new URL(parsed.data);
 				displayURL = url.hostname.startsWith('www.') ? url.hostname.slice(4) : url.hostname;
 				return;
 			}
+			displayURL = null;
 		}
 	});
 </script>
 
-{#if nonNullish(kongSwapDApp)}
-	<ModalExpandableValues>
-		<ModalValue slot="list-header">
-			<svelte:fragment slot="label">{$i18n.swap.text.swap_provider}</svelte:fragment>
-
-			<svelte:fragment slot="main-value">
-				<div class="flex gap-2">
-					<div class="mt-1">
-						<Logo
-							src={kongSwapDApp.logo}
-							alt={replacePlaceholders($i18n.dapps.alt.logo, { $dAppName: kongSwapDApp.name })}
-						/>
+{#if nonNullish(swapDApp) && nonNullish(selectedProvider) && nonNullish($swapAmountsStore)}
+	<CollapsibleBottomSheet showContentHeader>
+		{#snippet contentHeader()}
+			<ModalValue>
+				<svelte:fragment slot="label">
+					<div class="flex justify-center gap-2">
+						{$i18n.swap.text.swap_provider}
 					</div>
-					<div class="mr-auto">
-						<div class="text-lg font-bold">{kongSwapDApp.name}</div>
-						{#if nonNullish(displayURL)}
-							<div class="text-sm text-tertiary">{displayURL}</div>
+				</svelte:fragment>
+
+				<svelte:fragment slot="main-value">
+					<div class="flex items-start gap-3">
+						{#if isBestRate && $swapAmountsStore?.swaps.length > 1}
```

Comment:
your are checking for nullish already , no?

```suggestion
						{#if isBestRate && $swapAmountsStore.swaps.length > 1}
```